queue/QueueRunner.feature

```diff
@@ -16,69 +16,73 @@ Feature: Command and control using a message broker
-    And the client should not publish any response
+      | method       | call             |
+      | sum          | add two numbers  |
+      | increment    | increment number |
+    Then the client should consume all requests
+    And the client should publish the following responses:
+      | payload                             |
+      | {"result":3,"error":null,"id":"X1"} |
+      | {"result":4,"error":null,"id":"X2"} |
 
-  Scenario: Process messages then publish and stop
+  Scenario: Process messages then publish them all
     Given I receive the following requests:
       | payload                                        |
       | {"method":"sum","params":[1,2],"id":"X1"}      |
       | {"method":"increment","params":[3],"id":"X2"}  |
     When I go live with the following processing rules:
-      | method       | call             | action            |
-      | sum          | add two numbers  | publish and stop  |
-      | increment    | increment number | publish           |
-    Then the client should consume first request
+      | method       | call             |
+      | sum          | add two numbers  |
+      | increment    | increment number |
+    Then the client should consume all requests
     And the client should publish the following responses:
       | payload                             |
       | {"result":3,"error":null,"id":"X1"} |
+      | {"result":4,"error":null,"id":"X2"} |
 
 
   #  Display
-
+  
   Scenario: Display requests and responses
     Given I receive the following requests:
       | payload                                        |
       | {"method":"sum","params":[1,2],"id":"X1"}      |
       | {"method":"increment","params":[3],"id":"X2"}  |
     When I go live with the following processing rules:
-      | method       | call             | action            |
-      | sum          | add two numbers  | publish           |
-      | increment    | increment number | publish and stop  |
+      | method       | call             |
+      | sum          | add two numbers  |
+      | increment    | increment number |
     Then the client should display to console:
       | output                                 |
       | id = X1, req = sum(1, 2), resp = 3     |
       | id = X2, req = increment(3), resp = 4  |
-
-  Scenario: Display label next to unpublished responses
+  
+  Scenario: Display published response
     Given I receive the following requests:
       | payload                                        |
       | {"method":"sum","params":[1,2],"id":"X1"}      |
     When I go live with the following processing rules:
-      | method       | call             | action            |
-      | sum          | add two numbers  | stop              |
+      | method       | call             |
+      | sum          | add two numbers  |
     Then the client should display to console:
-      | output                                              |
-      | id = X1, req = sum(1, 2), resp = 3, (NOT PUBLISHED) |
```

Comment:
The label is still a valid use case. This is what we should display if we encounter an exception.

Reply:
You are right, in the absence of "stop" that code path is not exercised.